src/registry2.0/registry.tcl

```diff
@@ -236,7 +236,33 @@ proc open_file_map {args} {
 
 proc file_registered {file} {
 	global macports::registry.format
-	return [${macports::registry.format}::file_registered $file]
+
+	# Assume the file actually exists.
+	set cs true
+	set lowercase_file [string tolower $file]
+
+	# Check if the lower-case version exists.
+	if {[::file exists $lowercase_file]} {
+		# Just knowing that a lower-case version exists is not sufficient
+		# to come to the conclusion that the FS is case-insensitive.
+		# Both files *might* be unique files that just "happen"
+		# to be similar-named.
+		# Look at the inode number to distinguish these files.
+		if {![catch {::file lstat $file lstatret_orig}]} {
+			if {![catch {::file lstat $lowercase_file lstatret_lower}]} {
+				if {$lstatret_orig(ino) == $lstatret_lower(ino)} {
+					# File systems not supporting inodes, like FAT,
+					# will also land here. Luckily, FAT is mostly cis...
+					# If necessary, we could implement further checks
+					# based upon [file system], but refrain from this
+					# for now.
+					set cs false
```

Comment:
Not quite sure what to do here. We could add the function to pextlib and export it to Tcl, and just call it from here (and then just have it return 1 on !macOS)